init.lua

```diff
@@ -64,203 +49,73 @@ function letters.register_letters(modname, subname, from_node, description, tile
+				def.description = description.. " " ..char
+			elseif group == "u" then
+				def.description = description.. " " ..char:upper()
+			elseif group == "l" then
+				def.description = description.. " " ..char
+			end
+			def.inventory_image = tiles.. "^" ..tile.. "^[makealpha:255,126,126"
+			def.wield_image = def.inventory_image
+			def.tiles = {def.inventory_image}
 
-	for _, row in ipairs(letters) do
-
-		def = table.copy(def)
-		def.description = description.. " " ..row[3]
-		def.inventory_image = tiles.. "^letters_" ..row[1].. "_overlay.png^[makealpha:255,126,126"
-		def.wield_image = def.inventory_image
-		def.tiles = {def.inventory_image}
-
-		minetest.register_node(":" ..modname..":"..subname.. "_letter_" ..row[1],def)
-
-		def = table.copy(def)
-		def.description = description.. " " ..row[4]
-		def.inventory_image = tiles.. "^letters_" ..row[2].. "_overlay.png^[makealpha:255,126,126"
-		def.wield_image = def.inventory_image
-		def.tiles = {def.inventory_image}
-
-		minetest.register_node(":" ..modname..":"..subname.. "_letter_" ..row[2], def)
-
-		--[[minetest.register_craft({
-			output = from_node,
-			recipe = {
-				{modname..":"..name, modname..":"..name, modname..":"..name},
-				{modname..":"..name, modname..":"..name, modname..":"..name},
-				{modname..":"..name, modname..":"..name, modname..":"..name},
-			},
-		})--]]
+			minetest.register_node(":" ..modname..":"..subname.. "_letter_" ..char..group,def)
+		end
 	end
 	letter_cutter.known_nodes[from_node] = {modname, subname}
 end
 
 local cost = 0.110
 
-letter_cutter.names_lower = {
-	{"letter_al"},
-	{"letter_bl"},
-	{"letter_cl"},
-	{"letter_dl"},
-	{"letter_el"},
-	{"letter_fl"},
-	{"letter_gl"},
-	{"letter_hl"},
-	{"letter_il"},
-	{"letter_jl"},
-	{"letter_kl"},
-	{"letter_ll"},
-	{"letter_ml"},
-	{"letter_nl"},
-	{"letter_ol"},
-	{"letter_pl"},
-	{"letter_ql"},
-	{"letter_rl"},
-	{"letter_sl"},
-	{"letter_tl"},
-	{"letter_ul"},
-	{"letter_vl"},
-	{"letter_wl"},
-	{"letter_xl"},
-	{"letter_yl"},
-	{"letter_zl"},
-}
-
-letter_cutter.names_upper = {
-	{"letter_au"},
-	{"letter_bu"},
-	{"letter_cu"},
-	{"letter_du"},
-	{"letter_eu"},
-	{"letter_fu"},
-	{"letter_gu"},
-	{"letter_hu"},
-	{"letter_iu"},
-	{"letter_ju"},
-	{"letter_ku"},
-	{"letter_lu"},
-	{"letter_mu"},
-	{"letter_nu"},
-	{"letter_ou"},
-	{"letter_pu"},
-	{"letter_qu"},
-	{"letter_ru"},
-	{"letter_su"},
-	{"letter_tu"},
-	{"letter_uu"},
-	{"letter_vu"},
-	{"letter_wu"},
-	{"letter_xu"},
-	{"letter_yu"},
-	{"letter_zu"},
-}
-
-function letter_cutter:get_output_inv_lower(modname, subname, amount, max)
-
-	local list = {}
-	if amount < 1 then
-		return list
-	end
-
-	for i, t in ipairs(letter_cutter.names_lower) do
-		table.insert(list, modname .. ":" .. subname .. "_" .. t[1]
-			.. " " .. math.min(math.floor(amount/cost), max))
-	end
-	return list
-end
-
-function letter_cutter:get_output_inv_upper(modname, subname, amount, max)
+function letter_cutter:get_output_inv(modname, subname, amount, max, group)
```

Comment:
While you're at it, would you please be so nice to add comments where it makes sense? For example, what this function and its parameters do.